common/src/main/java/com/wynntils/services/translation/TranslationService.java

```diff
@@ -27,8 +31,13 @@ public void reloadData() {
         CachingTranslationProvider.loadTranslationCache();
     }
 
-    public void shutdown() {
-        CachingTranslationProvider.saveTranslationCache();
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldState.WORLD)
+            ;
```

Comment:
Huh?

Reply:
It was meant to be a return. Saving on any world state change, except world, as saving when joining both can cause slight lag. Also, it is theoretically impossible to have new, unsaved translations when joining a new world, as we've saved it before..